File: typescript/tests/payment/integration/paymentRequestIntegrationCommand.test.ts
import { describe, expect, it } from 'vitest';
import { Money } from '@/common/domain/primitive/money.js';
import { paymentRequestIntegrationCommand } from '@/payment/integration/paymentRequestIntegrationCommand.js';

describe('PaymentRequestIntegrationCommand', () => {
  it('creates a command with valid values', () => {
    const clientReference = crypto.randomUUID();
    const amount = new Money(50);

    const command = paymentRequestIntegrationCommand(clientReference, amount);

    expect(command.clientReference).toBe(clientReference);
    expect(command.amount).toBe(amount);
  });

  it('throws on null clientReference', () => {
    expect(() => paymentRequestIntegrationCommand(null as unknown as string, new Money(50))).toThrow();
  });

  it('throws on null amount', () => {
    expect(() => paymentRequestIntegrationCommand(crypto.randomUUID(), null as unknown as Money)).toThrow();
  });
});

File: java/src/test/java/it/giannibombelli/wsc2026/payment/integration/PaymentRequestIntegrationCommandTest.java
package it.giannibombelli.wsc2026.payment.integration;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class PaymentRequestIntegrationCommandTest {

    @Test
    void create_withValidValues_returnsCommand() {
        String clientReference = UUID.randomUUID().toString();
        Money amount = new Money(new BigDecimal("50.00"));

        PaymentRequestIntegrationCommand command = new PaymentRequestIntegrationCommand(clientReference, amount);

        assertThat(command.clientReference()).isEqualTo(clientReference);
        assertThat(command.amount()).isEqualTo(amount);
    }

    @Test
    void create_withNullClientReference_throwsException() {
        Money amount = new Money(new BigDecimal("50.00"));

        assertThatThrownBy(() -> new PaymentRequestIntegrationCommand(null, amount))
            .isInstanceOf(IllegalArgumentException.class);
    }

    @Test
    void create_withNullAmount_throwsException() {
        String clientReference = UUID.randomUUID().toString();

        assertThatThrownBy(() -> new PaymentRequestIntegrationCommand(clientReference, null))
            .isInstanceOf(IllegalArgumentException.class);
    }
}
